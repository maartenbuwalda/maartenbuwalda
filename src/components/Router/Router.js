import React from 'react'
import { Route, Switch } from 'react-router-dom'
import PropTypes from 'prop-types'
import Page from '../Page'
import Card from '../Card'
import { routes } from '../../constants/routes'
import { subPages } from '../../constants/subPages'

const Router = ({ position }) => {
  const navigation = [...routes, ...subPages]
  return (
    <Card position={position}>
      <Switch>
        {navigation.map((route, i) => {
          return <Route key={i} exact path={route.pathname} render={() => <Page page={route} />}/>
        })}
      </Switch>
    </Card>
  )
}

Router.propTypes = {
  position: PropTypes.string,
}

export default Router
