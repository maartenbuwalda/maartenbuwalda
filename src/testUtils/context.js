const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
