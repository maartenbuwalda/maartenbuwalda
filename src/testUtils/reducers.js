export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
