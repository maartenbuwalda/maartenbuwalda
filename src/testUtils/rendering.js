import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
