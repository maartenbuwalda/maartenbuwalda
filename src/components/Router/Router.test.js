import React from 'react'
import Router from './Router'
import { componentRenders } from '../../testUtils/rendering'

jest.mock('../../constants/subPages', () => ({
  subPages: [
    {
      pathname: '/',
      label: 'About me',
      title: 'Maarten Buwalda',
      image: 'images.me',
      icon: 'face',
      content: 'AboutMe',
    },
    {
      pathname: '/education',
      label: 'Education',
      title: 'Education',
      image: 'images.books',
      icon: 'book',
      content: 'Education',
    }
  ]
}))

jest.mock('../../constants/routes', () => ({
  routes: [
    {
      pathname: '/',
      label: 'About me',
      title: 'Maarten Buwalda',
      image: 'images.me',
      icon: 'face',
      content: 'AboutMe',
    },
    {
      pathname: '/education',
      label: 'Education',
      title: 'Education',
      image: 'images.books',
      icon: 'book',
      content: 'Education',
    }
  ]
}))

describe('Router component', () => {
  const component = <Router />
  componentRenders(component)
})
