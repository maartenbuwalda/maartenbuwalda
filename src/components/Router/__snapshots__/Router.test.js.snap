// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component renders correctly 1`] = `
<Card
  content=""
>
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      key="1"
      path="/education"
      render={[Function]}
    />
    <Route
      exact={true}
      key="2"
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      key="3"
      path="/education"
      render={[Function]}
    />
  </Switch>
</Card>
`;
